packages/opentelemetry-api/test/noop-implementations/noop-tracer.test.ts

```diff
@@ -20,7 +20,7 @@ import { Context } from '@opentelemetry/scope-base';
 
 describe('NoopTracer', () => {
   it('should not crash', () => {
-    const spanContext = { traceId: '', spanId: '' };
+    const spanContext = { traceId: '', spanId: '', traceFlags: 0 };
```

Comment:
in description the traceFlags is defined as `0x0` or `0x1` but here is just `0`,
can you use dedicated enum for that `TraceFlags.UNSAMPLED` ?

Reply:
Sure will adapt.
But honestly speaking I think that type `TraceFlags` for `spanContext.traceFlags` is actually wrong. Currently it's ok as there is just a single flag specified but once there are more flags in W3C spec it's problematic.

Reply:
so you saying that `traceFlags` of span context is not the same as `TraceFlags` type ? I thought it's the same, but if not the case then  maybe there should be a new enum and then use it here, wdyt ?

Reply:
Just looked once again into how typescript handles enums in detail and it seems the type is ok. But we should rename `UNSAMPLED` to `NONE` to avoid that people check for `UNSAMPLED`.

Will do this later together with the other updates.

Reply:
It also encourages checking in the form `if (traceFlags === TraceFlags.SAMPLED)` which will be problematic if new flags are added. I already previously corrected all instances of this, but it is an easy to miss error.